tests/python/pants_test/backend/python/rules/test_resolve_requirements.py

```diff
@@ -0,0 +1,52 @@
+# coding=utf-8
+# Copyright 2019 Pants project contributors (see CONTRIBUTORS.md).
+# Licensed under the Apache License, Version 2.0 (see LICENSE).
+
+from __future__ import absolute_import, division, print_function, unicode_literals
+
+import json
+import zipfile
+
+from pants.backend.python.rules.resolve_requirements import (ResolveRequirementsRequest,
+                                                             ResolvedRequirementsPex,
+                                                             )
+from pants.util.collections import assert_single_element
+from pants_test.test_base import TestBase
+
+
+class TestResolveRequirements(TestBase):
+
+  def create_pex_and_open_pex_info(
+    self, requirements=None, entry_point=None, interpreter_constraints=None
+  ):
+    def hashify_optional_collection(iterable):
+      return tuple(sorted(iterable)) if iterable is not None else tuple()
+
+    request = ResolveRequirementsRequest(
+      requirements=hashify_optional_collection(requirements),
+      output_filename="test.pex",
+      entry_point=entry_point,
+      interpreter_constraints=hashify_optional_collection(interpreter_constraints),
+    )
+    assert_single_element(
+      self.scheduler.product_request(ResolvedRequirementsPex, [request])
+    )
+    with zipfile.ZipFile("test.pex", "r") as pex:
```

Comment:
How could I go from a `Digest` (directory digest) to an actual file path? `DirectoryDigestToMaterialize`?

Reply:
There is an intrinsic to get the `Snapshot` for a `Digest`, so `x = yield Get(Snapshot, Digest(..))`.

To call that from a test, you would `...scheduler.product_request(Snapshot, [Digest(..)])`.